gc/gc-base/pom.xml

```diff
@@ -0,0 +1,122 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (C) 2020 Dremio
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.projectnessie</groupId>
+    <artifactId>nessie-gc</artifactId>
+    <version>0.15.2-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>nessie-gc-base</artifactId>
+
+  <name>Nessie - GC - Base Implementation</name>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.immutables</groupId>
+      <artifactId>value</artifactId>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.projectnessie</groupId>
+      <artifactId>nessie-client</artifactId>
+      <version>${project.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.projectnessie</groupId>
+      <artifactId>nessie-versioned-persist-in-memory</artifactId>
+      <version>${project.version}</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.projectnessie</groupId>
+      <artifactId>nessie-versioned-persist-in-memory</artifactId>
+      <version>${project.version}</version>
+      <type>test-jar</type>
+      <scope>test</scope>
+    </dependency>
```

Comment:
Could the dependencies be reversed so that each db-specific module depends on the GC module and runs GC tests for its dedicated backend (I think database adapter tests do that already)?.. as opposed to having the GC module depend on all backend modules... :thinking:

Reply:
yeah agree, the gc is independent of backend store. I will just keep for in-memory.

Reply:
Agree that they work differently. but the REST API output of fetching records from store is already tested right? So, this feature don't have to worry about backend store and can only focus on gc of the contents ?

Reply:
Unit tests of GC logic can probably be done with the in-memory adapter as a backing / mock store.

My concern was that if we want total coverage of integration tests for GC across all backends, would it not be preferable to have backend modules (many) depends on the GC module (one) and run the test code inherited from it on the local storage technology?